packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -92,8 +142,23 @@ export class Wallet extends EventEmitter<ObjectiveProposed> {
         try {
           const createResult = await this._engine.createChannel(p);
 
-          const {newObjective, channelResult} = createResult;
+          const assetHolders = createResult.channelResult.allocations.map(a =>
+            makeAddress(a.assetHolderAddress)
+          );
+          const listener = createChainListener(
+            this._engine,
+            this._engine.store,
+            this.messageService
+          );
+          this._chainService.registerChannel(
+            createResult.channelResult.channelId,
+            assetHolders,
+            listener
+          );
 
+          const {newObjective, channelResult, chainRequests} = createResult;
+          const transactions = await this._chainService.handleChainRequests(chainRequests);
+          await Promise.all(transactions.map(tr => tr.wait()));
```

Comment:
Are we sure that we want to wait for transaction mining before resolving this API call?

Reply:
That's a good point! For merging reasons I've updated this upstream in 9496f3a65